packages/aa/src/index.js

```diff
@@ -371,8 +404,24 @@ function comparePackageLogicalPaths(aPath, bPath) {
 }
 
 /**
- * @typedef Resolver
- * @property {(path: string, opts: { basedir: string }) => string} sync
+ * @template T
+ * @param {number} parallelMax
+ * @param {T[]} items
+ * @param {(item: T) => Promise<any>} mapper
+ * @returns {Promise<void>}
+ */
+async function runParallelMax(parallelMax, items, mapper) {
```

Comment:
note: this function could be much more sophisticated so that it tracks which async jobs finished and starts new ones as they do. The jobs are similar enough to each other tho, that it might not bring a lot of improvement to the execution time.

Reply:
update - quick test of the concept:
```js
async function runParallelMax(parallelMax, items, mapper) {
  items = items.reverse()
  return new Promise((resolve, reject) => {
    let promises = 0
    const boop = () => {
      if (items.length === 0) {
        if(promises === 0) resolve()
        return
      }
      const p = mapper(items.pop())
      promises++
      p.then(()=>{
        promises--
        boop()
      }, reject)
    }
    for (let i = 0; i < parallelMax; i++) {
      boop()
    }
  })
}
```
with this implementation it's a few percent slower, so there's not much to gain from smarter parallelization